fang/src/asynk/async_queue/async_queue_tests.rs

```diff
@@ -0,0 +1,321 @@
+use super::AsyncQueueable;
+use crate::asynk::AsyncRunnable;
+use crate::FangError;
+use crate::Scheduled;
+use async_trait::async_trait;
+use chrono::DateTime;
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct AsyncTask {
+    pub number: u16,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl AsyncRunnable for AsyncTask {
+    async fn run(&self, _queueable: &mut dyn AsyncQueueable) -> Result<(), FangError> {
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct AsyncUniqTask {
+    pub number: u16,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl AsyncRunnable for AsyncUniqTask {
+    async fn run(&self, _queueable: &mut dyn AsyncQueueable) -> Result<(), FangError> {
+        Ok(())
+    }
+
+    fn uniq(&self) -> bool {
+        true
+    }
+}
+
+#[derive(Serialize, Deserialize)]
+pub(crate) struct AsyncTaskSchedule {
+    pub number: u16,
+    pub datetime: String,
+}
+
+#[typetag::serde]
+#[async_trait]
+impl AsyncRunnable for AsyncTaskSchedule {
+    async fn run(&self, _queueable: &mut dyn AsyncQueueable) -> Result<(), FangError> {
+        Ok(())
+    }
+
+    fn cron(&self) -> Option<Scheduled> {
+        let datetime = self.datetime.parse::<DateTime<Utc>>().ok()?;
+        Some(Scheduled::ScheduleOnce(datetime))
+    }
+}
+
+/// This macro creates a module with tests for a `Queueable` type.
+///
+/// Arguments:
+/// + `$mod`: Name for the module
+/// + `$q`: Full path to type that implements `AsyncQueueable`
+/// + `$e`: An expression that returns a value of `$q` suitable for testing.
+///   + Multiple values returned by `$e` must be able to be interacted with concurrently without interfering with each other.
+macro_rules! test_asynk_queue {
```

Comment:
if we will have multiple back-ends, will we be able to know for which backed tests are being executed and tests of which back-end are failing?

Reply:
yes, this is determined by the module name. So we will pass `postgres` so it will create a module called postgres and we know from which backend is the possible error.

Reply:
check this line 


https://github.com/ayrat555/fang/pull/133/files#diff-0863d59c653afd9ecef465757a02de8079562074ed45a2cd95bbc0e5adf42149R749

Reply:
```rust
test_asynk_queue! {postgres, crate::AsyncQueue<bb8_postgres::tokio_postgres::tls::NoTls>, crate::AsyncQueue::<bb8_postgres::tokio_postgres::tls::NoTls>::test()}
```

Reply:
Yes, since the tests for each backend are generated in their own separate module, and `cargo test` shows the full paths of tests.